feature/s3/transfermanager/api_client.go

```diff
@@ -0,0 +1,47 @@
+package transfermanager
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws"
+)
+
+const userAgentKey = "s3-transfer"
+
+// DefaultMaxUploadParts is the maximum allowed number of parts in a multi-part upload
+// on Amazon S3.
+const DefaultMaxUploadParts int32 = 10000
+
+// DefaultPartSizeBytes is the default part size when transferring objects to/from S3
+const DefaultPartSizeBytes int64 = 1024 * 1024 * 8
+
+// DefaultMultipartUploadThreshold is the default size threshold in bytes indicating when to use multipart upload.
+const DefaultMultipartUploadThreshold int64 = 1024 * 1024 * 16
```

Comment:
[nit] I don't think this has to be typed to `int64`

Reply:
It doesn't, none of these constants do, compiler is supposed to figure it out in context of usage